test/jdk/java/foreign/TestAddressDereference.java

```diff
@@ -184,18 +186,23 @@ enum LayoutKind {
             this.layout = segment;
         }
 
+        private static final Pattern LAYOUT_PATTERN = Pattern.compile("^(?<align>\\d+%)?(?<char>[azcsifjdAZCSIFJD])\\d+$");
```

Comment:
On 32-bit platforms, the string representation of e.g. `long` is `4%j8`. This pattern helps to filter out the `4%`.